tracer.go

```diff
@@ -26,22 +26,7 @@ type Tracer interface {
 	// Like `StartTrace`, but the returned `Span` is made a child of `parent`.
 	//
 	// The `parent` parameter can either be a `context.Context` or an
-	// `opentracing.TraceContext`. In the former case, the implementation
-	// attempts to extract an `opentracing.Span` using `SpanFromGoContext()`.
+	// `opentracing.Span`. In the former case, the implementation attempts to
+	// extract an `opentracing.Span` using `SpanFromContext()`.
 	JoinTrace(operationName string, parent interface{}) Span
```

Comment:
looks like we don't need this method anymore. If parent is a context, devs can do
```go
SpanFromContext(context).StartChild(...)
```
and if it's already a span then just call `span.StartChild()`